procrastinate/worker.py

```diff
@@ -131,6 +132,9 @@ def run_job(self, job: jobs.Job) -> None:
         logger.info("Starting job", extra={"action": "start_job", "job": log_context})
         try:
             task_result = task(**job.task_kwargs)
+            if asyncio.iscoroutinefunction(task.func):
```

Comment:
Why not `asyncio.iscoroutine(task_result)` instead? That way we can handle regular functions returning coroutines.

Reply:
Realized the same thing when writing the doc, and I though it would be a nice follow up ticket in the future, but now that you've said it out loud, I guess I have no choice :D

Reply:
Done